src/util/SpawnPed.cpp

```diff
@@ -1,48 +1,79 @@
-#include "core/frontend/Notifications.hpp"
-#include "game/backend/FiberPool.hpp"
-#include "game/backend/ScriptMgr.hpp"
-#include "game/rdr/Natives.hpp"
-
+#include "SpawnPed.hpp"
 
 namespace YimMenu
 {
-	bool SpawnPed(std::string model_name, Player player, int playerPed)
+	extern int SpawnPed(std::string model_name, bool blockNewPedMovement, bool spawnDead, bool invincible, bool invisible, int scale)
 	{
-		FiberPool::Push([=] {
-			Hash model = MISC::GET_HASH_KEY(model_name.c_str());
-			if (STREAMING::IS_MODEL_IN_CDIMAGE(model) && STREAMING::IS_MODEL_VALID(model))
+		Hash model = MISC::GET_HASH_KEY(model_name.c_str());
+		if (STREAMING::IS_MODEL_IN_CDIMAGE(model) && STREAMING::IS_MODEL_VALID(model))
+		{
+			for (int i = 0; i < 5 && !STREAMING::HAS_MODEL_LOADED(model); i++)
+			{
+				STREAMING::REQUEST_MODEL(model, false);
+				ScriptMgr::Yield();
+			}
+
+			Vector3 coords = ENTITY::GET_OFFSET_FROM_ENTITY_IN_WORLD_COORDS(PLAYER::PLAYER_PED_ID(), 0.0, 3.0, -0.3);
```

Comment:
Instead of fetching a location in the function, set the coords as a param. That way you can ensure this function can be used anywhere. And don't we have like a namespace dedicated to our local values such as Self::PlayerPed?

Reply:
Yes, We have Self::PlayerPed. I will implement these fixes today.

Reply:
Just have the coords as a param to this function.

Reply:
Okay I will